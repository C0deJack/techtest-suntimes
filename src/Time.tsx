import styled from 'styled-components';

const StyledSection = styled.section`
    h2 {
        font-size: 1.5rem;
        font-weight: 500;
    }

    img.icon {
        width: 100px;
    }

    time {
        display: block;
        font-size: 2rem;
        font-weight: 700;
        text-transform: lowercase;
    }

    @media (min-width: 576px) {
        img.icon {
            width: 150px;
        }
    }
`;

interface Props {
    title: string;
    icon: string;
    time: string;
}

const Time = ({ title, icon, time }: Props): JSX.Element => {
    const trimmedTime = time.slice(0, -6);
    const amOrPm = time.slice(-2);

    return (
        <StyledSection>
            <img className='icon' src={icon} />
            <time>
                {trimmedTime}
                {amOrPm}
            </time>
            <h2>{title}</h2>
        </StyledSection>
    );
};

export default Time;
