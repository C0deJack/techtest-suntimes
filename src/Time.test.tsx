import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Time from './Time';

const testProps = {
    timeInput: '4:45:58 AM',
    timeOutput: '4:45AM',
    title: 'testTitle',
    titleRole: 'testTitle time',
    icon: 'testIconUrl',
};

test('Time component snapshot', () => {
    const time = renderer
        .create(
            <Time
                title={testProps.title}
                time={testProps.timeInput}
                icon={testProps.icon}
            />
        )
        .toJSON();
    expect(time).toMatchSnapshot();
});

describe('Time output', () => {
    beforeEach(() => {
        render(
            <Time
                title={testProps.title}
                time={testProps.timeInput}
                icon={testProps.icon}
            />
        );
    });

    test('Time renders the correct title', () => {
        const time = screen.getByText(/testTitle/i);
        expect(time).toBeInTheDocument();
    });

    test('Time renders the correct time', () => {
        const time = screen.getByRole('time', {
            name: new RegExp(testProps.title, 'i'),
        });
        expect(time).toBeInTheDocument();
        expect(time).toHaveTextContent(testProps.timeOutput);
    });
});
