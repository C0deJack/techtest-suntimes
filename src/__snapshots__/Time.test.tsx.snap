// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time component snapshot 1`] = `
<section
  className="sc-bdvvtL ffMMTR"
>
  <img
    className="icon"
    src="testIconUrl"
  />
  <time
    aria-label="testTitle"
    role="time"
  >
    4:45
    AM
  </time>
  <h2>
    testTitle
  </h2>
</section>
`;
